### DISCUSSION TITLE
Does GSLP support multiple diagram languages?

### DISCUSSION THREAD

**Discussion by amiths89** [eclipse-glsp/glsp#1285] [General] [answered]
I tried creating a second diagram language following the template in glsp-examples/project-templates/java-emf-theia but on running the glsp-client I get an ambiguous error with GLSPLayoutCommandContribution. I tried disabling layout, copy-paste and navigation using the flags which seemed to make the ambiguous error go away but my diagrams does not seem to come up and I can see another error related to SchemaContribution. Is there an alternative solution for a new diagram that follows the structure of the task editor example(similar to the the template java-emf-theia).

The following are the steps I followed:
glsp-client:
Created a copy of files and folders under tasklist-glsp and tasklist-theia and configured the path to these folders wherever applicable. While using the same port 5007 that was used by TaskList.

glsp-server
Created a separate model and a copy of all files except for the launcher code. 

Launcher
Configured the newly added copy of TaskListDiagramModule with the same server module used for TaskList.

**Comment by amiths89** [comment]
After this I tried again from the beginning following the steps mentioned here https://github.com/eclipse-glsp/glsp/discussions/1167#discussioncomment-7648466 . And it's pretty much the same. Can anyone let me know what exactly we would need to change in the TaskList example so as to be able to use 2 language extensions

**Reply by tortmayr** [reply, marked as answer]
@amiths89 Should be anwsered with https://github.com/eclipse-glsp/glsp/discussions/840
Please let me know if there are other issues.

**Reply by amiths89** [reply]
Oh this is perfect. Thanks a lot @tortmayr . I can now see where i was going wrong with folder structure.

**Reply by amiths89** [reply]
While this works perfectly fine when running individually I did notice as I refresh the page with both the diagrams opened in the workspace there seems to be an issue. It looks like the diagram type could be getting mixed up with the refresh when both the diagrams are open in the workspace. Is there some thing else that would need to be changed with regard to this diagram type or atleast while disposing the page.

**Reply by tortmayr** [reply]
Seems like the examples are still running on GLSP 2.0.0 which had a [bug] related to the handling of multiple simultaneously open diagrams. This has been  fixed with GLSP 2.1.0. I have created an issue for updating the example repos https://github.com/eclipse-glsp/glsp/issues/1297.

**Reply by amiths89** [reply]
@tortmayr If i were to incorporate the glsp 2.1.0 into my code, what all changes would I have to make. I updated the pom file but it looks that did not fix the issue so am guessing there are other changes to be made as well to migrate to glsp 2.1.0
